dd-trace-core/src/main/java/datadog/trace/core/flare/TracerFlarePoller.java

```diff
@@ -68,23 +71,24 @@ public Runnable register(Config config, SharedCommunicationObjects sco) {
     @Override
     public void accept(ConfigKey configKey, byte[] content, PollingRateHinter hinter)
         throws IOException {
-      if (configKey.getConfigId().startsWith(FLARE_LOG_LEVEL_PREFIX)) {
-        AgentConfigLayer agentConfigLayer =
-            AGENT_CONFIG_LAYER_ADAPTER.fromJson(
-                Okio.buffer(Okio.source(new ByteArrayInputStream(content))));
-        if (null != agentConfigLayer
-            && null != agentConfigLayer.config
-            && null != agentConfigLayer.config.logLevel) {
-          prepareForFlare(agentConfigLayer.config.logLevel);
-        } else {
-          cleanupAfterFlare();
-        }
+      AgentConfigLayer agentConfigLayer =
+          AGENT_CONFIG_LAYER_ADAPTER.fromJson(
+              Okio.buffer(Okio.source(new ByteArrayInputStream(content))));
+      if (null != agentConfigLayer
+          && null != agentConfigLayer.config
+          && null != agentConfigLayer.config.logLevel) {
+        configAction.put(configKey.getConfigId(), FLARE_LOG_LEVEL);
+        prepareForFlare(agentConfigLayer.config.logLevel);
+      } else {
+        cleanupAfterFlare();
       }
     }
 
     @Override
     public void remove(ConfigKey configKey, PollingRateHinter hinter) {
-      if (configKey.getConfigId().startsWith(FLARE_LOG_LEVEL_PREFIX)) {
+      String configId = configKey.getConfigId();
+      if (configAction.get(configId).equals(FLARE_LOG_LEVEL)) {
+        configAction.remove(configId);
```

Comment:
This could be simplified to:
```suggestion
      if (configAction.remove(configKey.getConfigId(), FLARE_LOG_LEVEL)) {
```

https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#remove-java.lang.Object-java.lang.Object-